File: sdks/guides/python/integrations/ocr_google_vision.py
# Third-party integration example - requires: pip install google-cloud-vision
# See: /guides/document-processing

from google.cloud import vision

client = vision.ImageAnnotatorClient()


# > Google Vision usage
def parse_document(content: bytes) -> str:
    image = vision.Image(content=content)
    response = client.document_text_detection(image=image)
    return response.full_text_annotation.text if response.full_text_annotation else ""
# !!

File: examples/typescript/guides/integrations/ocr-google-vision.ts
// Third-party integration - requires: pnpm add @google-cloud/vision
// See: /guides/document-processing

import { ImageAnnotatorClient } from '@google-cloud/vision';

const client = new ImageAnnotatorClient();

// > Google Vision usage
export async function parseDocument(content: Buffer): Promise<string> {
  const [result] = await client.documentTextDetection({ image: { content } });
  return result.fullTextAnnotation?.text ?? '';
}
